pytato/array.py

```diff
@@ -642,21 +667,28 @@ class DictOfNamedArrays(Mapping[str, Array]):
         This container deliberately does not implement
         arithmetic.
     """
+    _mapper_method = "map_dict_of_named_arrays"
 
     def __init__(self, data: Dict[str, Array]):
+        self._named_arrays = {name: NamedArray(self, name)
+                              for name in data}
         self._data = data
```

Comment:
```suggestion
        # kept around mainly for efficient __hash__
        self._data = data
```
(if that's true... otherwise `_data` is pretty redundant with `_named_arrays`.)